tests/integration/test_github.py

```diff
@@ -72,6 +97,15 @@ def test_create_gist(self):
         assert isinstance(g, github3.gists.Gist)
         assert g.public is True
 
+    def test_create_gpg_key(self):
+        """Test the ability of a GitHub instance to create a new GPG key."""
+        self.token_login()
+        cassette_name = self.cassette_name('create_gpg_key')
+        with self.recorder.use_cassette(cassette_name):
+            gpg_key = self.gh.create_gpg_key(GPG_KEY)
```

Comment:
In the past we've been a bit sloppy. Going forward I've started writing tests like so:

```
def test_create...(self):
    self.token_login()
    cassette_name = self.cassette_name('...')
    with self.recorder.use_cassette(...):
        gpg_key = self.gh.create_gpg_key(GPG_KEY)
        assert isinstance(gpg_key, github3.users.GPGKey)
        assert gpg_key.delete()
```

This way there's no out-of-band, manual clean-up required. Then for the


```
def test_gpg_key(self):
    ...
    with ...:
        created_gpg_key = self.gh.create_gpg_key(...)
        assert isinstance(created_gpg_key, ...)
        retrieved_gpg_key = self.gh.gpg_key(created_gpg_key.id)
        assert isinstance(retrieved_gpg_key, ...)
        assert created_gpg_key == retrieved_gpg_key
        assert created_gpg_key.delete(0
```

And the same goes for the last method too.

Reply:
Uh, I like the idea of having each test responsible for its cleanup. Another benefit is that anybody can re-record the cassette if need arises. Done!